var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -388,6 +388,10 @@ def install(self, spec, prefix):
         else:
             options.append('--with-clanguage=C')
 
+        # openmp is usually provided by the compiler and not a library dependency
+        if spec.satisfies('+openmp'):
+            options.append('--with-openmp=1')
```

Comment:
To protect against changes in defaults (or automatic configure time defaults) it's better to always explicitly specify the option:
```suggestion
        options.append('--with-openmp={0:d}'.format(1 if '+openmp' in spec else 0))
```

Reply:
@melven does this change work for you?

Reply:
I'm ok with the current patch - as its in the same format as the rest of the code. I guess at some point the code should be updated to this suggested format uniformly

Reply:
Ah there is code further ahead(with a different format) but does use the  suggested mode of setting either 1/0

So perhaps this fix can be moved there.
```diff
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
+            '--with-openmp=%s' % ('1' if '+openmp' in spec else '0'),
             '--with-64-bit-indices=%s' % ('1' if '+int64' in spec else '0')
```

Reply:
Sorry for the delay - I moved it as suggested by balay!